File: app/src/main/java/com/univreview/view/contract/SearchContract.java
package com.univreview.view.contract;

import com.univreview.adapter.contract.SearchAdapterContract;
import com.univreview.fragment.AbsListFragment;

/**
 * Created by DavidHa on 2017. 7. 15..
 */

public interface SearchContract {
    interface View{
        void setResult(int page);
        void setStatus(AbsListFragment.Status status);
    }

    void attachView(View view);
    void detachView();
    void setSubjectType(String type);
    void setSearchAdapterModel(SearchAdapterContract.Model model);
    void searchUniversity(String name, int page);
    void searchDepartment(long id, String name, int page);
    void searchMajor(long id, String name, int page);
    void searchProfessor(Long departmentId, String name, int page);
    void searchSubject(Long majorId, String name, int page);
    void searchProfFromSubj(long subjectId, String name, int page);
}

File: app/src/main/java/com/univreview/view/contract/SearchContract.kt
package com.univreview.view.contract

import com.univreview.fragment.AbsListFragment

/**
 * Created by DavidHa on 2017. 8. 6..
 */
interface SearchContract {
    interface View : BaseView {
        fun setResult(page: Int)
        fun setStatus(status: AbsListFragment.Status)
        fun setInputStr(searchStr : String)
    }

    fun callSearchApi(name: String, page: Int)
    fun stopSearch()
}